pkg/server/controllers.go

```diff
@@ -888,6 +889,65 @@ func (s *Server) installVirtualWorkspaceURLsController(ctx context.Context, conf
 	})
 }
 
+func (s *Server) installKubeQuotaController(
+	ctx context.Context,
+	config *rest.Config,
+	server *genericapiserver.GenericAPIServer,
+) error {
+	controllerName := "kcp-kube-quota-controller"
+	config = rest.AddUserAgent(config, controllerName)
+
+	kubeClusterClient, err := kubernetes.NewClusterForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	// TODO(ncdc): should we make these configurable?
+	const (
+		quotaResyncPeriod        = 5 * time.Minute
+		replenishmentPeriod      = 12 * time.Hour
```

Comment:
am curious, does this means that every 12h we might have a storm? Could one smear that over time for different workspaces?

Reply:
It would be based on when the workspace was "seen" by the controller. So for a long-running kcp, it would be spread based on when workspaces were created. But at startup, presumably it would see all the workspaces in a short time span, and would probably be more stormy. Follow-up issue ok?